File: 프로그래머스/체육복/training.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {
    int answer = 0;
    vector<int> v(31, 1);
    for (int i = 0; i < lost.size(); i++) {
        v[lost[i]] -= 1;
    }
    for (int i = 0; i < reserve.size(); i++) {
        v[reserve[i]] += 1;
    }
    for (int i = 1; i <= n; i++) {
        if (v[i] == 0) {
            if (i == 1) {
                if (v[i + 1] == 2) {
                    answer += 1;
                    v[i + 1] -= 1;
                }
            }
            else if (i == n) {
                if (v[i - 1] == 2) {
                    answer += 1;
                    v[i - 1] -= 1;
                }
            }
            else {
                if (v[i - 1] == 2) {
                    answer += 1;
                    v[i - 1] -= 1;
                }
                else if (v[i + 1] == 2) {
                    answer += 1;
                    v[i + 1] -= 1;
                }
            }
        }
        else answer++;
    }
    return answer;
}
int main() {

}

File: 프로그래머스/체육복/training.java
package com.company;
import java.util.Arrays;
import java.util.Collections;
import java.util.HashSet;
class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        int answer = n;
        Arrays.sort(lost);
        HashSet<Integer> set = new HashSet<>();
        for(int i=0; i<reserve.length;i++){
            set.add(reserve[i]);
        }
        for(int i=0; i<lost.length;i++){
            if(set.contains(lost[i])){
                set.remove(lost[i]);
            }
            else if(set.contains(lost[i]-1)){
                set.remove(lost[i]-1);
            }
            else if(i!=lost.length-1 && set.contains(lost[i]+1) && lost[i+1] != lost[i]+1){
                set.remove(lost[i]+1);
            }
            else
                answer--;
        }
        return answer;
    }
}
public class Main {
    public static void main(String[] args) {
        Solution a = new Solution();

    }
}
